scripts/check-docs/src/main.rs

```diff
@@ -0,0 +1,329 @@
+    line_no: usize,
+    name: String,
+    file: PathBuf,
+}
+
+#[allow(dead_code)]
+enum TestEnum {
+    Anchor(Vec<Anchor>),
+    Include(Vec<Include>),
+    Errors(Vec<Error>),
+}
+
+fn extract_starts_and_ends(
+    text_w_anchors: &str,
+) -> anyhow::Result<(Vec<Anchor>, Vec<Anchor>), Error> {
+    let apply_regex = |regex: Regex| {
+        text_w_anchors
+            .lines()
+            .filter_map(|line| regex.captures(line))
+            .map(|capture| {
+                let file = PathBuf::from(&capture[1]).canonicalize()?;
+                let line_no = &capture[2];
+                let anchor_name = &capture[3];
+
+                Ok(Anchor {
+                    line_no: line_no.parse().unwrap(),
+                    name: anchor_name.to_string(),
+                    file,
+                })
+            })
+            .collect::<Result<Vec<_>, Error>>()
+    };
+
+    let begins = apply_regex(Regex::new(
+        r"^(.+):(\d+):\s*(?:/{2,}|/\*)\s*ANCHOR\s*:\s*([\w_-]+)\s*(?:\*/)?",
+    )?)?;
+    let ends = apply_regex(Regex::new(
+        r"^(.+):(\d+):\s*(?:/{2,}|/\*)\s*ANCHOR_END\s*:\s*([\w_-]+)\s*(?:\*/)?",
+    )?)?;
+
+    Ok((begins, ends))
+}
+
+fn search_for_patterns_in_project(pattern: &str) -> anyhow::Result<String> {
+    let grep_project = std::process::Command::new("grep")
+        .arg("-I")
+        .arg("-H")
+        .arg("-R")
+        .arg("-n")
+        .arg("--exclude-dir=scripts")
+        .arg(pattern)
+        .arg(".")
+        .output()
+        .expect("failed grep command");
+
+    if !grep_project.status.success() {
+        bail!("Failed running grep command for searching {}", pattern);
+    }
+
+    Ok(String::from_utf8(grep_project.stdout)?)
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        extract_starts_and_ends, filter_valid_anchors, parse_includes, validate_includes, TestEnum,
+    };
+    use anyhow::bail;
+
+    fn test_search_for_patterns(pattern: &str, test_file: &str) -> anyhow::Result<String> {
+        let grep = std::process::Command::new("grep")
+            .arg("-I")
+            .arg("-H")
+            .arg("-R")
+            .arg("-n")
+            .arg("--exclude-dir=scripts")
+            .arg(pattern)
+            .arg(test_file)
+            .output()
+            .expect("failed grep command");
+
+        if !grep.status.success() {
+            bail!("Failed running grep command for searching {}", pattern);
+        }
+
+        Ok(String::from_utf8(grep.stdout)?)
+    }
+
+    fn contains_any(vec: &TestEnum, str: &str) -> bool {
+        match vec {
+            TestEnum::Anchor(anchor_vec) => anchor_vec.iter().any(|anchor| anchor.name == str),
+            TestEnum::Include(include_vec) => {
+                include_vec.iter().any(|include| include.anchor_name == str)
+            }
+            TestEnum::Errors(err_vec) => err_vec.iter().any(|err| err.to_string().starts_with(str)),
+        }
+    }
+
+    #[test]
+    fn test_anchors() -> anyhow::Result<()> {
```

Comment:
This is also an integration test, tying everything together. Per [convention](https://doc.rust-lang.org/rust-by-example/testing/integration_testing.html), you might want to separate it.